src/PatternBuilder.ts

```diff
@@ -592,6 +592,8 @@ export const oneOf: (
  */
 export const empty: Pattern = { tag: 'atom', kind: 'character', char: '' }
 
+// TODO @jacob-alford 22-11-13 Figure out how to test lines 602-604.  It's technically not unreachable during recursion
```

Comment:
it is unreachable: line 608, we don't recurse if `restMin.length === 0`.

unfortunately I don't think it's possible to specify "string of length n" in typescript (without questionable stuff)

Reply:
the conditions and padding enforced in `integerRange` (no underscore) make it so `restMin` and `restMax` are always the same length

Reply:
Oh gotcha.  I also tried removing it, and all the tests passed (with full coverage).  Do you think we could just remove it then?

Reply:
I think I was getting a linter warning or something because it doesn't know it can't be undefined there. but if it works, sure

Reply:
I'll double check tonight!